java/src/com/ibm/streamsx/topology/generator/spl/Preprocessor.java

```diff
@@ -110,6 +224,64 @@ private void relocateHashAdder(JsonObject hashAdder){
             // set Unparallel's output port schema using HashAdder's schema
             GraphUtilities.setOutputPortType(parent, 0, schema);
             GraphUtilities.setInputPortType(parent, 0, schema);
+        } else {
+            // Scenario #3: the parent $Unparallel$ has multiple children
+
+            // distinguish HashAdders from other children
+            LinkedList<JsonObject> HashAdders = new LinkedList<>();
```

Comment:
Why use a LinkedList?

Reply:
Because I don't know how many of the children are `HashAdder`s. What do you suggest?

Reply:
`ArrayList` - you don't need to know the size up-front, in most cases there will not be a resize.

`LinkedList` is used where there's need to remove individual items efficiently.